ThreadGroup/Controllers/TGMainViewController.m

```diff
@@ -199,9 +203,13 @@ - (void)configureMainViewForViewState:(TGMainViewState)viewState {
             break;
         case TGMainViewStateConnectDevicePassphrase:
         case TGMainViewStateConnectDeviceScanning:
-            [self resetSelectDeviceView];
-            [self setPopupNotificationForState:viewState animated:YES];
-            break;
+            if ([self isTogglingBetweenScanningAndPassphraseViewStates]) {
```

Comment:
I was also having the problem where the view was resetting when you were going from no wifi to wifi states, so I put up a quick fix for that in TGRootVC, which can now be deleted.

```
- (void)configureUIForReachableState {
    if (![self.childNavigationController.viewControllers containsObject:self.mainViewController]) {
        [self.mainViewController setViewState:TGMainViewStateLookingForRouters];
        [self.childNavigationController pushViewController:self.mainViewController animated:YES];
    } else {
        if (!(self.mainViewController.viewState == TGMainViewStateConnectDevicePassphrase)) {
            [self.mainViewController setViewState:self.mainViewController.viewState];
        }
        [self.childNavigationController popToViewController:self.mainViewController animated:YES];
    }
}
```
can now be
```
- (void)configureUIForReachableState {
    if (![self.childNavigationController.viewControllers containsObject:self.mainViewController]) {
        [self.mainViewController setViewState:TGMainViewStateLookingForRouters];
        [self.childNavigationController pushViewController:self.mainViewController animated:YES];
    } else {
        [self.mainViewController setViewState:self.mainViewController.viewState];
        [self.childNavigationController popToViewController:self.mainViewController animated:YES];
    }
}
```